external/vcm/vcm/data_transform.py

```diff
@@ -77,6 +97,205 @@ def Q2_from_dQ2_pQ2(ds):
     return ds
 
 
+@register(
+    [
+        "Qm",
+        DELP,
+        DLW_SFC,
+        DSW_SFC,
+        DSW_TOA,
+        ULW_SFC,
+        ULW_TOA,
+        USW_SFC,
+        USW_TOA,
+        LHF,
+        SHF,
+        COL_T_NUDGE,
+    ],
+    ["Qm_flux", "implied_downward_radiative_flux_at_surface"],
+)
+def Qm_flux_from_Qm_tendency(
+    ds, rectify_downward_radiative_flux=True, include_temperature_nudging=True,
+):
+    """See https://github.com/ai2cm/explore/blob/master/oliwm/
+    2021-12-13-fine-res-in-flux-form/2021-12-13-fine-res-in-flux-form-proposal-v2.ipynb
+    for derivation of transform."""
+    Qm_flux = -vcm.mass_cumsum(ds["Qm"], ds[DELP], dim="z")
+    toa_rad_flux = ds[DSW_TOA] - ds[USW_TOA] - ds[ULW_TOA]
+    upward_surface_mse_flux = ds[LHF] + ds[SHF] + ds[USW_SFC] + ds[ULW_SFC]
+
+    # add TOA boundary condition
+    Qm_flux = Qm_flux.pad(z=(1, 0), constant_values=0.0)
+    Qm_flux += toa_rad_flux
+    if include_temperature_nudging:
+        Qm_flux += ds[COL_T_NUDGE]
```

Comment:
Is this a 3d field where the value at level z is the cumulative sum up to z?

Reply:
Qm_flux is 3D, `ds[COL_T_NUDGE]` is 2D (it is the column-integrated fine-res nudging tendency, which is an output of the fine-res loader).